Assets/Scripts/GameCharacter/GameCharacterAnimationController.cs

```diff
@@ -1,70 +0,0 @@
-using DG.Tweening;
-using System;
-using System.Collections;
-using System.Collections.Generic;
-using TMPro;
-using UnityEngine;
-
-public class GameCharacterAnimationController : GameCharacterComponent
-{
-
-    [SerializeField] TextMeshPro statusDisplay;
-    [SerializeField] MeshRenderer gunBarrelMeshRendered;
-    [SerializeField] MeshRenderer characterBodyMeshRendered;
-    [SerializeField] GameObject gunSmokeParticalEffect;
-    [SerializeField] Color attackAnimationColor, takeDamageAnimationColor;
-
-    GameCharacterController characterController;
-    Color gunBarrelColor,characterMaterial;
-   
-    private void Start()
-    {
-        gunSmokeParticalEffect.SetActive(false);
```

Comment:
This is a classic case. You real dependency here is a IGunSmokeEffect with maybe 2 methods like
- Begin()
- End()
Then your implementation will be a monobehaviour component